README.md

```diff
@@ -691,17 +691,19 @@ class MyCharm(ops.CharmBase):
     def _on_start(self, _):
         foo = self.unit.get_container('foo')
         proc = foo.exec(['ls', '-ll'])
-        stdout, _ = proc.wait_output()
+        stdout, _ = proc.wait_output("...")
         assert stdout == LS_LL
 
 
 def test_pebble_exec():
     container = scenario.Container(
         name='foo',
-        exec_mock={
-            ('ls', '-ll'):  # this is the command we're mocking
-                scenario.ExecOutput(return_code=0,  # this data structure contains all we need to mock the call.
-                                    stdout=LS_LL)
+        execs={
+            "list-directory": scenario.Exec(  # "list-directory" is an arbitrary tag for the command
+                ('ls', '-ll'):  # this is the command we're mocking
+                return_code=0,
+                stdout=LS_LL
+            ),
```

Comment:
This was based on the earlier plan that @benhoyt and I came up with for 7.0 where `State` components would generally be dicts. However, since we're looking at sets instead now, I think this would be better as:

```suggestion
        execs={
            scenario.Exec(
                ('ls', '-ll'):  # this is the command we're mocking
                return_code=0,
                stdout=LS_LL
            ),
        },
```

*However*, if we're going to adopt the "use the first command match" behaviour from Harness (it does seem like a bunch of charms are making use of this) then order matters, so it would actually need to be a list (or tuple):

```suggestion
        execs=[
            scenario.Exec(
                ('ls', '-ll'):  # this is the command we're mocking
                return_code=0,
                stdout=LS_LL
            ),
        ],
```

I'm no longer convinced that having mocked command encapsulated into the `Exec` object alongside the results makes sense. So it would go back to the original, just with the rename:

```suggestion
    # The command we're mocking.
    cmd = ('ls', '-ll')
    # The mocked result to provide
    result = scenario.Exec(return_code=0, stdout=LS_LL)
    container = scenario.Container(name='foo', execs={cmd: result})
```

Reply:
I think the last suggestion (back to mapping, rename only) makes sense but has a few downsides:
- all other objects are flat frozensets, why is this one a mapping?
- Dict is mutable. Perhaps we could use a frozendict to prevent further mutation at runtime? 

Re the 'people rely on definition order' issue: it's true that dicts are ordered nowadays, but still it's a bit smelly to rely on that behaviour IMHO.

I also miss the context of the harness exec design, but I think we could make our lives easier if we force people to use the full command they want to mock. 

We could also allow multiple commands per exec:

```python
        execs={
            scenario.Exec(
                cmds=(
                    ('ls'), 
                    ('ls', '-ll'), 
                ),
                return_code=0,
                stdout=LS_LL
            ),
        },
   
```

Reply:
@tonyandrewmeyer Let's discuss briefly after our daily sync. Given that we've settled on sets for everything, that's where I'd lean. Also, the "prefix matching" has been quite useful for Harness, though if that makes the implementation problematic, I'm open to not doing it.

Reply:
@benhoyt and I discussed this in a call and agree that since most of the other components are moving to frozensets, this should as well (which means moving the command back into the object). @benhoyt suggested using `command_prefix` rather than `command` (like Harness) to make it clearer that it's going to pattern match.

I think if you want multiple commands to have the same return code / stdout / stderr, then it's pretty simple to just put those in variables and use them when constructing multiple Execs (in a set comprehension, perhaps).